app.js

```diff
@@ -82,6 +87,32 @@ function validateArgs(url, opts, cb) {
     cb(new error.BackupError('InvalidOption', 'Invalid resume option, must be type boolean'), null);
     return;
   }
+
+  // Validate URL and ensure no auth if using key
```

Comment:
Is it worth checking the length of host/path? i.e. `urlObject.host.length !== 0`

Reply:
As well as `undefined` and `null` the empty string is `falsy` in javascript, so I don't think the length check adds anything.

Reply:
When I was looking at this though I noticed that the exception message was a little confusing so I updated it in 42af7450c99f47a337024f7359b99eb119fd68db

Reply:
Ok sounds good.